File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    unordered_map<int, int> hash;
    int spieces=0;
    for (int i=0; i<nums.size(); i++) {
        if (!hash[nums[i]]) spieces++;
        hash[nums[i]]=1;
    }
    
    int choiceNum = nums.size()/2;
    answer = min(spieces, choiceNum);
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    pick = len(nums) // 2
    nums = set(nums)
    # 중복을 제거했는데 처음보다 크거나 같다 --> 고를 수 있는 수는 pick으로 제한
    if len(nums) >= pick:
        answer = pick 
    else:
        answer = len(nums)
        
    return answer